### ISSUE TITLE
Support Firebase 8

### ISSUE THREAD

**Issue by AlexanderArvidsson** [prescottprue/redux-firestore#312]
**What is the current behavior?**
The import is done using `import * as firebase from "firebase/app";`.
Firebase 8.0 changes this to the default import, `import firebase from "firebase/app"`.
As such, there are typing errors.
```
Argument of type 'typeof firebase' is not assignable to parameter of type 'typeof import("/home/metamist/dev/app-next/web/node_modules/firebase/index")'.
  Property 'firebase' is missing in type 'typeof firebase' but required in type 'typeof import("/home/metamist/dev/app-next/web/node_modules/firebase/index")'.ts(2345)
```

**What is the expected behavior?**
No typing errors should occur, and the import should be changed to the default.

**Which version of redux-firestore are you using? What about other dependencies?**
redux-firestore 0.14.0

**Which environments/browsers are affected by this issue? Did this work in previous versions or setups?**
Using with firebase 8

**Minimal demo to reproduce issue (using [codesandbox](https://codesandbox.io/) or similar)**
No need, it is easy to figure out why this happens. The breaking change in Firebase 8 is that you should use the default import.

**Comment by prescottprue** [comment]
Thanks for reaching out

**Comment by AlexanderArvidsson** [comment]
Since in previous firebase versions, you have to import it as `import * as firebase ...`, that means making this will be a breaking change.
As such, if I were to make a pull request with these changes, you would also have to up your major version of this package and not support firebase 7 and below in the future.
I don't know if there's a way to somehow support both versions at once, but if there is a way that is much more preferable.

**Comment by prescottprue** [comment]
Yeah agreed - this will have to be in another major version where we will also have have to update the peer dep setting to require firebase ^8. If you make the PR I can handle the version change/release

**Comment by cabello** [comment]
Found this issue while trying to add appCheck to my app.
App was using v7, appCheck was introduced on v8, I followed the release notes: https://firebase.google.com/support/release-notes/js#version_800_-_october_26_2020

<img width="885" alt="Screen Shot 2022-01-21 at 8 19 56 AM" src="https://user-images.githubusercontent.com/60936/150533818-9e7ec7fb-ae51-4ac8-9b1d-f29b1ed8bfbe.png">

I will put a PR for this next following your suggestion.

**Comment by jowo-io** [comment]
Any progress on Firebase 9 support ?

**Comment by cabello** [comment]
The PR I put up should be enough for v8 support, v9 would need to be investigated, changelog etc, @jowo-io can you take care of that?

**Comment by jowo-io** [comment]
> import firebase from "firebase/app"

I'd be happy to help, adding support for v9 may be quite a large change, but I was able to get it working locally on v9 by importing the compat version of firebase
```
import firebase from "firebase/compat/app"
```
and then running
```
npm install --force
```
to skip the warnings thrown by `redux-firestore` regarding 
```
  "peerDependencies": {
    "firebase": "^4 || ^5 || ^6 || ^7 || ^8"
  },
```

**Comment by jowo-io** [comment]
i'm not currently familiar enough with the changes to firebase v9 to say much more at this point, but I'll do some reading and post back

**Comment by cabello** [comment]
Yeah the upgrade guide is quite complete, likely it will be a much larger PR: https://firebase.google.com/docs/web/modular-upgrade

**Comment by cabello** [comment]
@prescottprue would you mind taking a look at the PR please, it's one line change and it would unlock us moving from firebase 7 to 8. Thank you 🙇

**Comment by prescottprue** [comment]
Released in v1.0.0 - it seemed to break types with older versions. Thanks for the ping @cabello
